[service-bus] T2 - revert back to lazy-initialization model for .createSender() (#9302)
Azure/azure-sdk-for-js: sdk/servicebus/service-bus/migrationguide.md

  ruleManager.removeRule();
  ```

> * createSessionReceiver() is now an async method and initialize the connection

Reviewer: ```suggestion
* createSessionReceiver() is now an async method. The promise returned by this method
is resolved when a receiver link has been initialized with a session in the service.
```

Prior to v7 `createSessionReceiver()` worked using lazy-initialization, where the

> AMQP connection would only be initialized on first send or receiving of a message.

Reviewer: ```suggestion
the receiver link to the session is only initialized when any of the async methods on the `SessionReceiver` is first called
```

Author: Slight modification (I believe the tense should still be past, not present).

> The connection and link are now initialized after calling `createSessionReceiver`.

Reviewer: This line can be deleted if the above 2 suggested edits are made

```suggestion
```
